framework/components/ecl-list-items/ecl-list-items.scss

```diff
@@ -45,9 +45,41 @@
 }
 
 .ecl-list-item__primary {
+  flex-shrink: 0;
   height: auto;
   margin-right: map-get($ecl-spacing, 'm');
   max-height: 5.55rem;
   max-width: 5.55rem;
   width: auto;
 }
+
+// Highlited item
+.ecl-list-item--highlight .ecl-list-item__link {
```

Comment:
Would it be better to update the twig logic adding additional checks and simplifying the markup rather than going more specific and weak on the rules here?

Quickly checking the markup, there are more elements which are not used rather than used:

```twig
<li class="ecl-list-item ecl-list-item--highlight">
  <a href="" class="ecl-link ecl-list-item__link">
    <div class="ecl-u-sr-only">List item</div>
    <div class="ecl-list-item__primary">
      <img alt="State of the European Union 2017" src="../../assets/demo_images/nature-demo-1.jpg">
    </div>
    <div class="ecl-list-item__body">
      <!-- unused -->
      <div class="ecl-list-item__meta">
        <div class="ecl-meta ecl-meta--highlight">
        </div>
      </div>
      <h3 class="ecl-list-item__title ecl-heading ecl-heading--h3">State of the European Union 2017</h3>
      <!-- unused -->
      <p class="ecl-list-item__detail ecl-paragraph">
      </p>
      <!-- unused -->
      <div class="ecl-list-item__action">
      </div>
    </div>
  </a>
</li>
```

Reply:
Well, it is the same for other variants. At first I used a specific markup for each variant, but that ended in duplicated code. So I decided to use the same markup for all variants, and decide what to display in the context.
What could be improved here is to add tests in markup to not display empty html tags. Or we can do a specific markup for each variant, as you proposed.

To be discussed...

Reply:
Which-ever is more convenient for you.
Making some additional checks to avoid empty tags will be safer also in terms that it won't have breaking changes, so i'm ok with that and having the specificity if it's necessary.